### PR TITLE
Set API server URL from Kubernetes client for Fleet

### PR DESCRIPTION
**Pull request by weyfonk**
#42649 was submitted against `release/v2.8`, at a time when `release/v2.9` did not yet exist.

This PR is its 2.9 counterpart.
Refers to #44630.
See #41848 for more context.

### CODE REVIEW

File: pkg/fleet/fleet.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		return "", fmt.Errorf("fleet.GetClusterHost: %w", err)
+	}
+
+	if clientCfg == nil {
+		return fail(errors.New("client config not set"))
+	}
+
+	rawConfig, err := clientCfg.RawConfig()
+	if err != nil {
+		return fail(errors.New("could not convert client config into raw config"))
```
[the remark this answers is not in the document]
Comment (weyfonk) [reply pkg/fleet/fleet.go:28]: How would you see something like this?
```
@@ -16,7 +16,7 @@ func GetClusterHost(clientCfg clientcmd.ClientConfig) (string, error) {
        }
 
        fail := func(err error) (string, error) {
-               return "", fmt.Errorf("fleet.GetClusterHost: %w", err)
+               return "", fmt.Errorf("fleet.GetClusterHost: unable to determine cluster host: %w", err)
        }
 
        if clientCfg == nil {
@@ -25,7 +25,7 @@ func GetClusterHost(clientCfg clientcmd.ClientConfig) (string, error) {
 
        rawConfig, err := clientCfg.RawConfig()
        if err != nil {
-               return fail(errors.New("could not convert client config into raw config"))
+               return fail(fmt.Errorf("no configuration available: %w", err))
        }
 
        cluster, ok := rawConfig.Clusters[rawConfig.CurrentContext]
```

[the remark this answers is not in the document]
Comment (bigkevmcd) [reply pkg/fleet/fleet.go:28]: To be clear, I don't think this is a "must fix".

I'm just looking to simplify a bit :-)

File: pkg/fleet/fleet.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	cluster, ok := rawConfig.Clusters[rawConfig.CurrentContext]
+	if ok {
+		return cluster.Server, nil
+	}
+
+	for _, v := range rawConfig.Clusters {
```
[the remark this answers is not in the document]
Comment (weyfonk) [reply pkg/fleet/fleet.go]: Warning log added, as discussed. Great suggestion, thank you.

File: pkg/fleet/fleet.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (weyfonk) [reply pkg/fleet/fleet.go:29]: Right, I've committed the above, thinking that it makes error messages somewhat more user-readable.

**Review by bigkevmcd — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]

**Pull request merged**